Classes/Telemetry/BITChannel.m

```diff
@@ -230,19 +246,19 @@ - (void)startTimer {
   if ([self timerIsRunning]) {
     [self invalidateTimer];
   }
-  
+
   self.timerSource = dispatch_source_create(DISPATCH_SOURCE_TYPE_TIMER, 0, 0, self.dataItemsOperations);
   dispatch_source_set_timer(self.timerSource, dispatch_walltime(NULL, NSEC_PER_SEC * self.batchInterval), 1ull * NSEC_PER_SEC, 1ull * NSEC_PER_SEC);
   __weak typeof(self) weakSelf = self;
   dispatch_source_set_event_handler(self.timerSource, ^{
     typeof(self) strongSelf = weakSelf;
     if (strongSelf) {
-        if (strongSelf->_dataItemCount > 0) {
-            [strongSelf persistDataItemQueue];
-        } else {
-            strongSelf.channelBlocked = NO;
-        }
-        [strongSelf invalidateTimer];
+      if (strongSelf->_dataItemCount > 0) {
```

Comment:
@gwynne Barriers will not help [here](https://github.com/erychagov/HockeySDK-Mac/blob/86c09f208fa497407022ef6c679f5032c0620d9b/Classes/Telemetry/BITChannel.m#L192). `dataItemCount` is
```
An integer value that keeps tracks of the number of data items added to the JSON Stream string
```
So, it need to be changed together with `BITSafeJsonEventsString`. IMHO mutex prefered than 2CAS here. This is not crash handling code and mutex should works ok.